components/backup-stream/src/endpoint.rs

```diff
@@ -879,24 +903,36 @@ where
             let mts = self.prepare_min_ts().await;
             let sched = self.scheduler.clone();
             info!("min_ts prepared for flushing"; "min_ts" => %mts);
+            let (tx, _) = tokio::sync::mpsc::channel(1);
             self.region_op(ObserveOp::ResolveRegions {
                 callback: Box::new(move |res| {
-                    try_send!(sched, Task::ExecFlush(task, res));
+                    try_send!(sched, Task::ExecFlush(task, res, tx));
                 }),
                 min_ts: mts,
             })
             .await
         })
     }
 
-    fn on_exec_flush(&mut self, task: String, resolved: ResolvedRegions) {
+    fn on_exec_flush(&mut self, task: String, resolved: ResolvedRegions, cb: Sender<FlushResult>) {
         self.checkpoint_mgr.freeze();
-        self.pool
-            .spawn(root!("flush"; self.do_flush(task, resolved).map(|r| {
-                if let Err(err) = r {
-                    err.report("during updating flush status")
+        let fut = self.do_flush(task.clone(), resolved);
+        let syncer = self.checkpoint_mgr.make_syncer();
+        self.pool.spawn(root!("flush"; async move {
+            let res = fut.await;
+            if let Err(ref err) = &res {
+                err.report("during updating flush status")
+            }
+            // If nobody waits us, it is no need to construct the result.
+            if !cb.is_closed() {
+                let flush_res = FlushResult { task, error: res.err() };
+                if !syncer.await {
```

Comment:
Will this syncer works on extreme scenario?

Even do_flush flush finished. this only make the ` Task::RegionCheckpointsOp(RegionCheckpointOperation::FlushWith` send to task queue first.

if the task queue block for some reasons. the `manager_handle` will delay the FlushWith message. and then the `manager_handle ` from syncer seems come out firstly. which makes `make_syncer` not work any more.